File: api/src/main/java/dev/ulloasp/mlsuite/schema/application/dto/UpdateSchemaDraftRequest.java
package dev.ulloasp.mlsuite.schema.application.dto;

import java.util.List;
import java.util.Map;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;

public record UpdateSchemaDraftRequest(
        @jakarta.validation.constraints.NotNull Long expectedDraftRevision,
        @NotBlank String name,
        @NotEmpty Map<String, Object> formSchema,
        @NotEmpty List<Map<String, Object>> bindings) {
}

File: frontend/src/api/schemas/dtos/update-schema-draft-request.ts
/*
SPDX-License-Identifier: MIT
Copyright (c) 2025 Pablo Ulloa Santin
*/

import type { JsonRecord } from "./json-record";
import type { SchemaDraftBindingDto } from "./schema-draft-binding-dto";

export type UpdateSchemaDraftRequest = {
  expectedDraftRevision: number;
  name: string;
  formSchema: JsonRecord;
  bindings: SchemaDraftBindingDto[];
};
